internal/handlers/pg/msg_findandmodify.go

```diff
@@ -271,6 +271,9 @@ func (h *Handler) upsert(ctx context.Context, tx pgx.Tx, docs []*types.Document,
 			}
 		}
 
+		// we must place the _id to the first index of the upsert
+		upsert.MoveIDToTheFirstIndex()
```

Comment:
I see you call it for upsert but not for `insert` or `update`. Why? Wouldn't it be enough to call it `types.Document.ValidateData` instead?

Reply:
For insert or update operations it would be called inside MarshallJSON. I don't get the part for `types.Document.ValidateData`.

Reply:
@w84thesun it's a different check. If your doc doesn't have `_id` at all, it will come to this check. If this `if` doesn't exist, and your test doesn't have `_id`, the code below will be executed.

Reply:
@w84thesun I mean, why do we need to have `MoveIDToTheFirstIndex` in a few different places, if for all of them `Document.ValidateData` is called before insert or update. Can we just move this logic of id sorting there? So we ensure that it's sorted before we send it to the db. Or it's not enough?

Reply:
Also, `h.insert` is called on the line 277, and it executes the same thing as for `insert`. You say, the logic for `insert` is already in `MarshalJSON`, why do we need this extra line then?

Reply:
> Also, `h.insert` is called on the line 277, and it executes the same thing as for `insert`. You say, the logic for `insert` is already in `MarshalJSON`, why do we need this extra line then?

No need, removed it.